src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -140,25 +152,40 @@ def _call_api(self, url, payload, order_id=None):
 
         return response_json
 
-    def _get_form_token(self, request, order, billing_address, credit_card_token=None):
+    def _get_form_token(
+        self,
+        request,
+        order,
+        billing_address,
+        credit_card_token=None,
+        capture_only=False,
+    ):
         """
         Get the form token from the API
 
         https://docs.lyra.com/fr/rest/V4.0/api/playground/Charge/CreatePayment
         https://docs.lyra.com/fr/rest/V4.0/api/playground/Charge/CreateToken
         """
-        payment_data = self._get_payment_data(
-            request, order, billing_address, credit_card_token
+        payment_data = self._get_payload_data(
+            request, order, billing_address, credit_card_token, capture_only
         )
 
         url = f"{self.api_url}Charge/CreatePayment"
+        if capture_only:
+            url = f"{self.api_url}Charge/CreateToken"
```

Comment:
I think this method is doing to much thing. IMO you can split it to manage exception in one of them.